src/ImageSharp/IImageProcessingContext{TPixel}.cs

```diff
@@ -14,6 +14,12 @@ namespace SixLabors.ImageSharp
     public interface IImageProcessingContext<TPixel>
         where TPixel : struct, IPixel<TPixel>
     {
+        /// <summary>
+        /// Gets the source image bounds
+        /// </summary>
+        /// <returns>The <see cref="Rectangle"/></returns>
+        Rectangle Bounds();
```

Comment:
As far as I understand, this is always the boundary of the **initial** image.
What happens if someone applies multiple operations, with the sizes `sizeA > sizeB > sizeC` ?
```C#
image.Mutate(c => c.Resize(sizeA).Resize(sizeB).Resize(sizeB) );
```

Reply:
Good catch! The default implementation would have only worked when mutating not cloning. I've updated and added tests.

Reply:
I'm still a bit worried if this method is good for us:

We have added a visible state to an object that looked kinda stateless towards it's consumers. I remember we did not want to expose source/destination images on the interface for some reason, but actually, if we are exposing `Bounds`, it's almost the same. 

I understand the reason however, that you need this because the initialization logic lives now mostly inside the processor constructors, but maybe there are other solutions to achieve the same goals.

@tocsoft your thoughts?

If we are unsure, how about hiding this behind `IInternalImageProcessingContext<T>` to avoid exposing it to users?